File: src-tauri/src/application/dto/accounting.rs
//! Accounting DTOs - Simplified income/expense model

use serde::{Deserialize, Serialize};

/// Create accounting entry request
#[derive(Debug, Deserialize)]
pub struct CreateEntryRequest {
    pub date: String,
    pub entry_type: String, // "income" or "expense"
    pub category: String,   // "tuition", "rent", "salaries", "utilities", "other"
    pub description: String,
    pub amount: f64,
    pub reference: Option<String>,
}

/// Accounting entry response
#[derive(Debug, Serialize)]
pub struct AccountingEntryDto {
    pub id: String,
    pub date: String,
    pub entry_type: String,
    pub category: String,
    pub description: String,
    pub amount: f64,
    pub reference: Option<String>,
    pub created_at: String,
}

impl From<crate::domain::entities::accounting::AccountingEntry> for AccountingEntryDto {
    fn from(entry: crate::domain::entities::accounting::AccountingEntry) -> Self {
        Self {
            id: entry.id,
            date: entry.date,
            entry_type: entry.entry_type.as_str().to_string(),
            category: entry.category.as_str().to_string(),
            description: entry.description,
            amount: entry.amount,
            reference: entry.reference,
            created_at: entry.created_at,
        }
    }
}

/// Accounting summary for dashboard
#[derive(Debug, Serialize)]
pub struct AccountingSummaryDto {
    pub total_income: f64,
    pub total_expenses: f64,
    pub net_balance: f64,
    pub entry_count: i64,
    pub recent_entries: Vec<AccountingEntryDto>,
    pub monthly_data: Vec<MonthlyDataPointDto>,
    pub expenses_by_category: Vec<CategoryTotalDto>,
    pub income_by_category: Vec<CategoryTotalDto>,
}

/// Monthly data point for charts
#[derive(Debug, Serialize)]
pub struct MonthlyDataPointDto {
    pub month: String,
    pub income: f64,
    pub expenses: f64,
}

/// Category total for breakdown
#[derive(Debug, Serialize)]
pub struct CategoryTotalDto {
    pub category_name: String,
    pub amount: f64,
}

File: src/features/accounting/types/accounting.ts
// Accounting Types - Simplified
// Mirrors simplified Rust DTOs

export const INCOME_CATEGORY = {
  TUITION: "tuition",
  OTHER: "other",
} as const;

export type IncomeCategory = (typeof INCOME_CATEGORY)[keyof typeof INCOME_CATEGORY];

export const EXPENSE_CATEGORY = {
  RENT: "rent",
  SALARIES: "salaries",
  UTILITIES: "utilities",
  OTHER: "other",
} as const;

export type ExpenseCategory = (typeof EXPENSE_CATEGORY)[keyof typeof EXPENSE_CATEGORY];

export type AccountingEntryType = "income" | "expense";

export interface AccountingEntry {
  id: string;
  date: string;
  entry_type: AccountingEntryType;
  category: string;
  description: string;
  amount: number;
  reference?: string;
  created_at: string;
}

export interface AccountingSummary {
  total_income: number;
  total_expenses: number;
  net_balance: number;
  entry_count: number;
  recent_entries: AccountingEntry[];
  monthly_data: MonthlyDataPoint[];
  expenses_by_category: ExpenseByCategory[];
  income_by_category: IncomeByCategory[];
}

export interface MonthlyDataPoint {
  month: string;
  income: number;
  expenses: number;
}

export interface ExpenseByCategory {
  category_name: string;
  amount: number;
}

export interface IncomeByCategory {
  category_name: string;
  amount: number;
}

export interface CreateEntryPayload {
  date: string;
  category: string;
  description: string;
  amount: number;
}

export interface EntryFilters {
  date_from?: string;
  date_to?: string;
  type?: AccountingEntryType;
}
